### PR TITLE
[Merged by Bors] - feat(FieldTheory/Minpoly): add theorem for minpoly of -x

### PR DESCRIPTION
**Pull request by jjdishere**
Add theorem `minpoly A (- x) = (-1) ^ (natDegree (minpoly A x)) * (minpoly A x).comp (- X)`

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Polynomial/RingDivision.lean [hunk 1]
```diff
@@ -548,6 +548,26 @@ theorem rootMultiplicity_mul' {p q : R[X]} {x : R}
...
+    (p.comp (-X)).leadingCoeff = (-1) ^ p.natDegree * p.leadingCoeff := by
+  nontriviality R
+  by_cases h : p = 0
+  · simp [h]
+  rw [Polynomial.leadingCoeff, natDegree_comp_eq_of_mul_ne_zero, coeff_comp_degree_mul_degree] <;>
+  simp [mul_comm, h]
+
+theorem Monic.neg_one_pow_natDegree_mul_comp_neg_X {p : R[X]} (hp : p.Monic) :
+    ((-1) ^ p.natDegree * p.comp (-X)).Monic := by
+  simp [Monic]
```
Comment (jcommelin) [inline Mathlib/Algebra/Polynomial/RingDivision.lean:562]: Please replace this simp call with the output of `simp? [Monic]`. That is more robust.
(This is called "squeezing the simp".)

Non-squeezed simps are "flexible" tactics, and they can only be followed by other flexible tactics.

Comment (jjdishere) [reply Mathlib/Algebra/Polynomial/RingDivision.lean:562]: Thank you! Fixed

File: Mathlib/FieldTheory/Minpoly/Field.lean [hunk 2]
```diff
@@ -148,6 +148,27 @@ theorem sub_algebraMap {B : Type*} [CommRing B] [Algebra A B] {x : B}
...
+      have n2 := qmo.neg_one_pow_natDegree_mul_comp_neg_X.ne_zero
+      rw [degree_eq_natDegree qmo.ne_zero,
+        degree_eq_natDegree n1, natDegree_mul (by simp) (right_ne_zero_of_mul n1), natDegree_comp]
+      rw [degree_eq_natDegree (minpoly.ne_zero hx),
+        degree_eq_natDegree qmo.neg_one_pow_natDegree_mul_comp_neg_X.ne_zero,
+        natDegree_mul (by simp) (right_ne_zero_of_mul n2), natDegree_comp] at H
+      simpa using H
+  · rw [minpoly.eq_zero hx, minpoly.eq_zero, zero_comp]
+    · simp only [natDegree_zero, pow_zero, mul_zero]
+    · exact fun h => hx (neg_neg x ▸ IsIntegral.neg h)
```
Comment (Ruben-VandeVelde) [inline Mathlib/FieldTheory/Minpoly/Field.lean:170]: It feels like we should have `IsIntegral.of_neg` / `isIntegral_neg_iff`

Comment (jjdishere) [reply Mathlib/FieldTheory/Minpoly/Field.lean:170]: Yes. I'll add it.

Comment (jjdishere) [reply Mathlib/FieldTheory/Minpoly/Field.lean:170]: Added.

**Review by Ruben-VandeVelde — APPROVED** [review]
Thanks! 

maintainer merge

**Comment by riccardobrasca** [comment]
Thanks!

bors merge

**Pull request closed without merging**